docs/parameters.md

```diff
@@ -0,0 +1,322 @@
+`valueFrom` and `type: json` fields allows to declare this as well:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is JSON which should be 
+      # validated and passed as a JSON object
+      type: json
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: password
+```
+
+#### Other primitive types (int, bool, float)
+
+```yaml
+  ...
+  parameters:
+    - name: text
+      value: 1 # string by default, will be sent as "1"
+    - name: number
+      type: int
+      value: 1 # will be sent without quotes
+    - name: flag
+      type: bool
+      value: true
+    - name: fl
+      type: float
+      value: 123.5
+```
+
+#### Arrays
+A separate field `array: true` can be used to declare a value as an array. The
+element type is defined by the `type` value as described above.
+
+```yaml
+  ...
+  parameters:
+    - name: tags
+      array: true
+      value:
+      - foo
+      - bar
+    - name: jsonArray
+      type: json
+      array: true
+      value: [{"a": 1, "b": 2}]
+```
+
+### Passing all parameters from a single source at once
+
+In addition to support for explicitly specifying every parameter value fetched from 
+a particular source, Service Catalog also supports passing multiple parameters at 
+once using a `parametersFrom` field (similar to how `envFrom` works for `Pod`s).
+Instead of specifying a reference to a single `Secret` (or `ConfigMap`) key, you 
+just set the reference to the `Secret` (or `ConfigMap`), and all keys will be 
+automatically processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
+```
+
+Parameter values could be passed as JSON objects, the same way as for [individual parameters](#json-object):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
+
+When a parameter name exists in multiple sources defined in `parametersFrom` or `parameters` 
+field, the value associated with the last source will take precedence.
+Values defined in `parameters` with a duplicate name will take precedence over `parametersFrom`.
```

Comment:
ok I think I understand what you're saying... and even though this might be consistent with how Pods work, yikes that's complicated :-)  

How about something like:
Within the scope the scope of a single `parameters` or `parametersFrom` block, if there are duplicate top-level names then the last one in each section will take precedent. If there are duplicate top-level names across `parameters` and `parametersFrom` blocks then the `parameters` takes precedent over `parametersFrom`.


What is there are more than one parameters or parametersFrom blocks?  Last wins?